src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -177,20 +184,48 @@ protected override SynthesizedAttributeData TrySynthesizeIsReadOnlyAttribute()
             return base.TrySynthesizeIsReadOnlyAttribute();
         }
 
+        protected override SynthesizedAttributeData TrySynthesizeIsByRefLikeAttribute()
+        {
+            if ((object)_lazyIsByRefLikeAttribute != null)
+            {
+                return new SynthesizedAttributeData(
+                    _lazyIsByRefLikeAttribute.Constructor,
+                    ImmutableArray<TypedConstant>.Empty,
+                    ImmutableArray<KeyValuePair<string, TypedConstant>>.Empty);
+            }
+
+            return base.TrySynthesizeIsByRefLikeAttribute();
+        }
+
         private void CreateEmbeddedAttributesIfNeeded(DiagnosticBag diagnostics)
         {
             if (this.NeedsGeneratedIsReadOnlyAttribute)
             {
-                CreateEmbeddedAttributeIfNeeded(
-                    ref _lazyEmbeddedAttribute,
-                    diagnostics,
-                    AttributeDescription.CodeAnalysisEmbeddedAttribute);
+                CreateEmbeddedAttributeItselfIfNeeded(diagnostics);
 
                 CreateEmbeddedAttributeIfNeeded(
                     ref _lazyIsReadOnlyAttribute,
                     diagnostics,
                     AttributeDescription.IsReadOnlyAttribute);
             }
+
+            if (this.NeedsGeneratedIsByRefLikeAttribute)
+            {
+                CreateEmbeddedAttributeItselfIfNeeded(diagnostics);
+
+                CreateEmbeddedAttributeIfNeeded(
+                    ref _lazyIsByRefLikeAttribute,
+                    diagnostics,
+                    AttributeDescription.IsByRefLikeAttribute);
+            }
+        }
+
+        private void CreateEmbeddedAttributeItselfIfNeeded(DiagnosticBag diagnostics)
```

Comment:
Why is that different than other attributes? why wouldn't be us then creating a `CreateXXXXattributeIfNeeded` for each type then?

---
In reply to: [118793343](https://github.com/dotnet/roslyn/pull/19795#discussion_r118793343) [](ancestors = 118793343,118792352)